dataplane/driver.go

```diff
@@ -297,6 +297,9 @@ func StartDataplaneDriver(configParams *config.Config,
 				InterfaceName:       configParams.WireguardInterfaceName,
 				MTU:                 configParams.WireguardMTU,
 				RouteSource:         configParams.RouteSource,
+				// currently, AKS *must* encrypt hostnetwork traffic, while it is optional on EKS
+				HostEncryptionEnabled: configParams.KubernetesProvider() == config.ProviderAKS ||
```

Comment:
I'm thinking `HostEncryptionEnabled` sounds too similar to the name of the flag, but struggling to come up with a better name. Maybe `EncryptHostTraffic` or `HostEncryptionSupported`? The reason is that someone might enable the `WireguardHostEncryptionEnabled` flag on say GKE which we don't support.

Reply:
Had a hunch about the same - I think `EncryptHostTraffic` sounds the clearest, will rewrite that name in a sec :+1: